plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/policies/FBNConnectionEndpointPolicy.java

```diff
@@ -120,14 +122,34 @@ public void layout() {
 			}
 		};
 		if (con.getSourceDecoration() != null) {
+			figure.add(selectonOverlayFigure(con.getSourceDecoration()));
```

Comment:
you may need to give the `figure` a stack layout so that both subfigures are drawn on top of each other.